uc-crux-llvm/src/UCCrux/LLVM/Context/Function.hs

```diff
@@ -117,24 +120,51 @@ ppFunctionContextError =
         ]
     BadMemType _ -> "Bad MemType"
 
+throwFunctionContextError :: FunctionContextError -> a
+throwFunctionContextError err =
+  case err of
+    BadMemType {} -> isMalformed
+    BadLift {} -> isMalformed
+    BadLiftArgs {} -> doPanic
+  where
+    nm = "throwFunctionContextError"
+    doPanic = panic nm [Text.unpack (ppFunctionContextError err)]
+    isMalformed =
+      malformedLLVMModule nm (PP.pretty (ppFunctionContextError err)) []
+
 withPtrWidthOf ::
   LLVMTrans.ModuleTranslation arch ->
   (ArchOk arch => b) ->
   b
 withPtrWidthOf trans k =
   LLVMTrans.llvmPtrWidth (trans ^. LLVMTrans.transContext) (\ptrW -> withPtrWidth ptrW k)
 
--- | This function does some precomputation of ubiquitously used values, and
--- some handling of what should generally be very rare errors.
+-- | This function does some precomputation of ubiquitously used values.
+--
+-- Any errors encountered in this process are bugs in UC-Crux or results of a
+-- malformed LLVM module, and are thrown as exceptions.
 makeFunctionContext ::
   forall m arch fullTypes.
   ArchOk arch =>
   ModuleContext m arch ->
   DefnSymbol m ->
   Ctx.Assignment (FullTypeRepr m) fullTypes ->
   Ctx.Assignment CrucibleTypes.TypeRepr (MapToCrucibleType arch fullTypes) ->
-  Either FunctionContextError (FunctionContext m arch fullTypes)
+  FunctionContext m arch fullTypes
 makeFunctionContext modCtx defnSymb argFullTypes argTypes =
+  case tryMakeFunctionContext modCtx defnSymb argFullTypes argTypes of
+    Left err -> throwFunctionContextError err
```

Comment:
I suggest putting `HasCallStack` constraints on `throwFunctionContextError` and all of the functions that call it, so that you get call stacks for free on panics.

Reply:
Okay, let me read a little bit about that and I'll add those constriants! Thanks for the suggestion.